src/Evolution/Systems/Cce/ScriPlusInterpolationManager.hpp

```diff
@@ -0,0 +1,607 @@
+ * `insert_target_time()`), multiplying the two interpolation results together
+ * before returning.
+ * Template parameters:
+ * - `ToInterpolate`: the vector type associated with the values to interpolate.
+ * - `::Tags::Multiplies<MultipliesLhs, MultipliesRhs>`: The tag associated with
+ * the interpolation procedure. This determines the behavior of the
+ * interpolation return value. The default is just interpolation, if `Tag` has
+ * prefix  `::Tags::Multiplies` (this case) or `Tags::Du`, the interpolator
+ * performs the additional multiplication or time derivative as a step in the
+ * interpolation procedure.
+ */
+template <typename ToInterpolate, typename MultipliesLhs,
+          typename MultipliesRhs>
+struct ScriPlusInterpolationManager<
+    ToInterpolate, ::Tags::Multiplies<MultipliesLhs, MultipliesRhs>> {
+ public:
+  ScriPlusInterpolationManager(
+      const size_t target_number_of_points, const size_t vector_size,
+      std::unique_ptr<intrp::SpanInterpolator> interpolator) noexcept
+      : interpolation_manager_lhs_{target_number_of_points, vector_size,
+                                   interpolator->get_clone()},
+        interpolation_manager_rhs_{target_number_of_points, vector_size,
+                                   std::move(interpolator)} {}
+
+  /// provide data to the interpolation manager.
+  ///
+  /// \details `u_bondi` is a vector of
+  /// inertial times, and `to_interpolate_lhs` and `to_interpolate_rhs` are the
+  /// vector of values that will be interpolated to target times. The
+  /// interpolation result will be the product of the interpolated lhs and rhs
+  /// vectors.
+  void insert_data(const DataVector& u_bondi,
+                   const ToInterpolate& to_interpolate_lhs,
+                   const ToInterpolate& to_interpolate_rhs) noexcept {
+    interpolation_manager_lhs_.insert_data(u_bondi, to_interpolate_lhs);
+    interpolation_manager_rhs_.insert_data(u_bondi, to_interpolate_rhs);
+  }
+
+  /// \brief Request a target time to be interpolated to when enough data has
+  /// been accumulated.
+  ///
+  /// For optimization, we assume that these are inserted in ascending order.
+  void insert_target_time(const double time) noexcept {
+    interpolation_manager_lhs_.insert_target_time(time);
+    interpolation_manager_rhs_.insert_target_time(time);
+  }
+
+  /// Determines whether enough data before and after the first time in the
+  /// target time queue has been provided to assemble an accurate full
+  /// interpolated vector.
+  bool first_time_is_ready_to_interpolate() const noexcept {
+    return interpolation_manager_lhs_.first_time_is_ready_to_interpolate() and
+           interpolation_manager_rhs_.first_time_is_ready_to_interpolate();
+  }
+
+  /// \brief Interpolate to the first targe time in the queue, returning both
+  /// the time and the interpolated and multiplied data at that time.
+  ///
+  /// note: This will work regardless of whether it is 'ready' to interpolate
```

Comment:
`\note`

If it is not able to interpolate to the full accuracy using a centered stencil, it will perform the best interpolation available.